dll/win32/shell32/CDefaultContextMenu.cpp

```diff
@@ -1341,6 +1342,29 @@ CDefaultContextMenu::InvokeRegVerb(
     return S_OK;
 }
 
+HRESULT
+CDefaultContextMenu::_DoInvokeCommandCallback(
+    LPCMINVOKECOMMANDINFOEX lpcmi, WPARAM CmdId)
+{
+    BOOL Unicode = (lpcmi->fMask & CMIC_MASK_UNICODE) && lpcmi->cbSize >= FIELD_OFFSET(CMINVOKECOMMANDINFOEX, ptInvoke);
+    WCHAR lParamBuf[MAX_PATH];
+    LPARAM lParam = 0;
+
+    if (Unicode && lpcmi->lpParametersW)
+        lParam = (LPARAM)lpcmi->lpParametersW;
+    else if (lpcmi->lpParameters)
+        lParam = SHAnsiToUnicode(lpcmi->lpParameters, lParamBuf, _countof(lParamBuf)) ? (LPARAM)lParamBuf : 0;
+
+    HRESULT hr;
+#if 0 // TODO: Try DFM_INVOKECOMMANDEX first.
+    DFMICS dfmics = { sizeof(DFMICS), lpcmi->fMask, lParam, ?, ?, (LPCMINVOKECOMMANDINFO)lpcmi, m_site };
+    hr = _DoCallback(DFM_INVOKECOMMANDEX, CmdId, &dfmics);
```

Comment:
Theoretically, if the DFM_INVOKECOMMANDEX mutates the contents of the `lpcmi`, but returns `E_NOTIMPL`, will the `DFM_INVOKECOMMAND` handling continue with the modified args?

Reply:
This is true yes but I don't see how the callback would do that by accident.